api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/KVariable.kt

```diff
@@ -19,17 +22,52 @@ import org.tensorflow.op.core.Variable
  * @property [name] name of the variable
  * @property [shape] shape of the variable
  * @property [variable] corresponding [Variable] object
- * @property [initializerOperand] variable initializer
+ * @property [initializerOperation] variable initializer
  * @property [regularizer] variable regularizer
  */
 public data class KVariable(
     val name: String,
     val shape: Shape,
     val variable: Variable<Float>,
-    val initializerOperand: Operand<Float>,
+    val initializerOperation: InitializerOperation,
     val regularizer: Regularizer?
 )
 
+/**
+ * A class that keeps information which allows to initialize [KVariable].
+ *
+ * @property [assign] assign operation for the variable
+ * @property [initialValue] value generated by the default initializer
+ */
+public data class InitializerOperation(val assign: Assign<Float>, val initialValue: Operand<Float>) {
```

Comment:
Will it support another type of assign operands, like assignandadd or something else?

Reply:
Are `AssignAdd` operations usable for variable initialization? Anyway, I can change the type to `Operand<Float>` (same as the initalizer type in `KVariable` before the refactoring).